warehouse/ingest-core/src/main/java/datawave/iterators/FrequencyColumnIterator.java

```diff
@@ -0,0 +1,179 @@
+            log.trace("Transforming range for key " + sortedKeyValueIterator.getTopKey().getRow().toString(), new Exception());
+        
+        Key topKey = null, aggregatedKey = null;
+        Value topValue = null;
+        Value aggregatedValue = null;
+        boolean isFrequencyRange = false;
+        
+        while (sortedKeyValueIterator.hasTop()) {
+            numRecords++;
+            Text cq = sortedKeyValueIterator.getTopKey().getColumnQualifier();
+            topKey = sortedKeyValueIterator.getTopKey();
+            topValue = sortedKeyValueIterator.getTopValue();
+            
+            // Just do an identity transform if there it is not a Frequency Column
+            if (!topKey.getColumnFamily().toString().equals(ColumnFamilyConstants.COLF_F.toString())) {
+                kvBuffer.append(sortedKeyValueIterator.getTopKey(), sortedKeyValueIterator.getTopValue());
+                sortedKeyValueIterator.next();
+                isFrequencyRange = false;
+                continue;
+            } else {
+                isFrequencyRange = true;
+            }
+            
+            if (cq.toString().startsWith(MetadataHelper.COL_QUAL_PREFIX)) {
+                newKey = topKey;
+                frequencyFamilyCounter.deserializeCompressedValue(topValue);
+                if (frequencyFamilyCounter.getDateToFrequencyValueMap().size() == 0)
+                    log.error("Compressed value was not deserialized properly");
+                aggregatedValue = topValue;
+                aggregatedKey = topKey;
+
+                if (log.isTraceEnabled())
+                    log.trace("Aggregate Key: " + aggregatedKey.toStringNoTime());
+                
+                if (!rowIdToCompressedFreqCQMap.containsKey(aggregatedKey.getRow() + cq.toString()))
+                    rowIdToCompressedFreqCQMap.put(aggregatedKey.getRow().toString() + cq.toString(), frequencyFamilyCounter);
+                else {
+                    FrequencyFamilyCounter previousCounter = rowIdToCompressedFreqCQMap.get(aggregatedKey.getRow() + cq.toString());
+                    for (Map.Entry<YearMonthDay,Frequency> entry : previousCounter.getDateToFrequencyValueMap().entrySet()) {
+                        if (ageOffDate.compareTo(entry.getKey().getYyyymmdd()) < 0)
+                            frequencyFamilyCounter.aggregateRecord(entry.getKey().getYyyymmdd(), String.valueOf(entry.getValue().getValue()));
+                    }
+                    rowIdToCompressedFreqCQMap.remove(aggregatedKey.getRow() + cq.toString());
+                    rowIdToCompressedFreqCQMap.put(aggregatedKey.getRow() + cq.toString(), frequencyFamilyCounter);
+                    
+                }
+                
+            } else {
+                
+                if (ageOffDate.compareTo(cq.toString().substring(cq.getLength() - 8)) < 0)
+                    frequencyFamilyCounter.aggregateRecord(cq.toString(), topValue.toString());
```

Comment:
Here is where the exception stems from that I was talking about this morning.  topValue.toString() is inappropriate here.  topValue contains an integer encoded as bytes, not as a string.  The value is ingested here: https://github.com/NationalSecurityAgency/datawave/blob/d9846a1057fd73944bd8626c4986e691a3194968/warehouse/ingest-core/src/main/java/datawave/ingest/metadata/EventMetadata.java#L415

Reply:
I recommend changing the aggregateRecord function to simply take in an integer or a long.  This avoid any string parsing along the way.

Reply:
Then this line would change to SummingCombiner.VAR_LEN_ENCODER.decode(topeValue.get()).intValue().